tests/Auth0Test.php

```diff
@@ -132,4 +132,37 @@ private function getAccessToken()
             ->disableOriginalConstructor()
             ->getMock();
     }
+
+    public function testGetAuthorizationUrlWithCustomDomain()
+    {
+        $customDomain = 'https://login.custom-domain.tld';
+        $provider = new OauthProvider(array_merge($this->config, ['customDomain' => $customDomain]));
+        $url = $provider->getAuthorizationUrl();
+
+        $this->assertTrue(0 === strpos($url, $customDomain));
+    }
+
+    /**
+     * Test that URL getters work as expected with custom domain set, and account not set.
+     * They should not throw AccountNotProvidedException (or any exception),
+     * and have to return an url starting with the custom domain.
+     */
+    public function testCustomDomain()
+    {
+        $customDomain = 'https://login.custom-domain.tld';
```

Comment:
IMO, the `$customDomain` should not contain the protocol. Maybe, the `domain` method should return only the domain and a new method `baseUrl` should be added and used by `getBaseAuthorizationUrl`, `getBaseAuthorizationUrl`, etc.